pallets/asset-handler/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+		/// Asset Withdrawn (recipient, ResourceId, Amount)
+		AssetWithdrawn(H160, ResourceId, BalanceOf<T>),
+		FeeUpdated(BridgeChainId, BalanceOf<T>, )
+	}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Migration is not operational yet
+		NotOperational,
+		/// MinterMustBeRelayer
+		MinterMustBeRelayer,
+		/// ChainIsNotWhitelisted
+		ChainIsNotWhitelisted,
+		/// NotEnoughBalance
+		NotEnoughBalance
+	}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+
+		/// Creates new Asset where AssetId is derived from chain_id and contract Address
+		///
+		/// # Parameters
+		///
+		/// * `chain_id`: Asset's native chain
+		/// * `contract_add`: Asset's actual address at native chain
+		#[pallet::weight(195_000_000)]
+		pub fn create_asset(
+			origin: OriginFor<T>,
+			chain_id: BridgeChainId,
+			contract_add: H160
+		) -> DispatchResult {
+			T::AssetCreateUpdateOrigin::ensure_origin(origin)?;
+			let rid = chainbridge::derive_resource_id(chain_id, &contract_add.0);
+			T::AssetManager::create(
+				Self::convert_asset_id(rid),
+				chainbridge::Pallet::<T>::account_id(),
+				true,
+				BalanceOf::<T>::one().unique_saturated_into(),
+			)?;
+			Self::deposit_event(Event::<T>::AssetRegistered(rid));
+			Ok(())
+		}
+
+        /// Mints Asset into Recipient's Account
+		/// Only Relayers can call it.
+		///
+		/// # Parameters
+		///
+		/// * `destination_add`: Recipient's Account
+		/// * `amount`: Amount to be minted in Recipient's Account
+		/// * `rid`: Resource ID
+		#[pallet::weight(195_000_000)]
+		pub fn mint_asset(origin: OriginFor<T>, destination_add: T::AccountId, amount: BalanceOf<T>, rid: ResourceId) -> DispatchResult{
+			let sender = ensure_signed(origin)?;
+			ensure!(chainbridge::Pallet::<T>::is_relayer(&sender), Error::<T>::MinterMustBeRelayer);
+			T::AssetManager::mint_into(Self::convert_asset_id(rid), &destination_add, amount.saturated_into::<u128>())?;
+			Self::deposit_event(Event::<T>::AssetDeposited(destination_add, rid, amount));
+			Ok(())
+		}
+
+		/// Transfers Asset to Destination Chain.
+		///
+		/// # Parameters
+		///
+		/// * `chain_id`: Asset's native chain
+		/// * `contract_add`: Asset's actual address at native chain
+		/// * `amount`: Amount to be burned and transferred from Sender's Account
+		/// * `recipient`: recipient
+		#[pallet::weight(195_000_000)]
+		pub fn withdraw(origin: OriginFor<T>, chain_id: BridgeChainId, contract_add: H160, amount: BalanceOf<T>, recipient: H160) -> DispatchResult{
+			let sender = ensure_signed(origin)?;
+			ensure!(chainbridge::Pallet::<T>::chain_whitelisted(chain_id), Error::<T>::ChainIsNotWhitelisted);
+			let rid = chainbridge::derive_resource_id(chain_id, &contract_add.0);
+			ensure!(T::AssetManager::reducible_balance(Self::convert_asset_id(rid), &sender, true)>=amount.saturated_into::<u128>(), Error::<T>::NotEnoughBalance);
```

Comment:
What about the existential deposit of that Asset?

Reply:
reducible_balance fn checks it internally. Basically, it returns (free_balance - existential_deposit).

Reply:
Where is the config for the existential deposit of an asset?

Reply:
https://github.com/Polkadex-Substrate/Polkadex/blob/zk-chainbridge-integration/pallets/asset-handler/src/lib.rs#L121